### PR TITLE
Support code coverage statistics

### PR DESCRIPTION
**Pull request by Linary**
Change-Id: If6d6e71feb4d6d0a7643fcb4e7d05c72d57d256e

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -47,7 +51,7 @@ jobs:
         - |
           if [ $DEPLOYED = 0 ]; then
             DEPLOYED=1
-            mvn cobertura:cobertura > /dev/null && bash <(curl -s https://codecov.io/bash)
+            bash <(curl -s https://codecov.io/bash)
```
Comment (javeme) [inline .travis.yml]: no longer needed, it is to run the test before

File: .travis.yml [hunk 2]
```diff
@@ -37,6 +38,9 @@ script:
       mvn test -P tinkerpop-process-test,$BACKEND
     fi
 
+after_script:
```
Comment (javeme) [inline .travis.yml]: prefer to report just if after_success

File: hugegraph-cassandra/src/main/java/com/baidu/hugegraph/backend/store/cassandra/CassandraTable.java [hunk 3]
```diff
@@ -210,7 +209,7 @@ protected void registerMetaHandlers() {
         List<Select> selects = new ArrayList<>(ids.size());
         for (List<Object> id : ids) {
             assert nameParts.size() == id.size();
-            Select idSelect = cloneSelect(select, this.table());
+            Select idSelect = QueryBuilder.select().from(this.table());
```
Comment (javeme) [inline hugegraph-cassandra/src/main/java/com/baidu/hugegraph/backend/store/cassandra/CassandraTable.java]: may loss field such as order-by

File: hugegraph-test/pom.xml [hunk 4]
```diff
@@ -12,11 +12,81 @@
...
+        <dependency>
+            <groupId>com.baidu.hugegraph</groupId>
+            <artifactId>hugegraph-mysql</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.baidu.hugegraph</groupId>
+            <artifactId>hugegraph-palo</artifactId>
+            <version>${project.version}</version>
+        </dependency>
```
Comment (javeme) [inline hugegraph-test/pom.xml]: add comments for why add these dependencies and why no hugegraph-api

File: hugegraph-test/pom.xml [hunk 5]
```diff
@@ -12,11 +12,81 @@
...
+                </exclusion>
+                <exclusion>
+                    <groupId>com.baidu.hugegraph</groupId>
+                    <artifactId>hugegraph-mysql</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>com.baidu.hugegraph</groupId>
+                    <artifactId>hugegraph-palo</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (javeme) [inline hugegraph-test/pom.xml]: remove

File: hugegraph-test/pom.xml [hunk 6]
```diff
@@ -35,134 +105,143 @@
...
+                    </filesets>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.jacoco</groupId>
+                <artifactId>jacoco-maven-plugin</artifactId>
+                <version>0.8.3</version>
+                <executions>
+                    <execution>
+                        <id>pre-unit-test</id>
```
Comment (javeme) [inline hugegraph-test/pom.xml]: pre-test

File: hugegraph-test/pom.xml [hunk 7]
```diff
@@ -35,134 +105,143 @@
...
+                        <goals>
+                            <goal>prepare-agent</goal>
+                        </goals>
+                    </execution>
+                    <execution>
+                        <id>post-unit-test</id>
```
Comment (javeme) [inline hugegraph-test/pom.xml]: post-test

File: hugegraph-test/pom.xml [hunk 8]
```diff
@@ -35,134 +105,143 @@
...
+                    </execution>
+                    <execution>
+                        <id>api-test</id>
+                        <configuration>
+                            <testSourceDirectory>${basedir}/src/main/java/
+                            </testSourceDirectory>
+                            <testClassesDirectory>${basedir}/target/classes/
+                            </testClassesDirectory>
+                            <includes>
+                                <include>**/ApiTestSuite.java</include>
```
Comment (javeme) [inline hugegraph-test/pom.xml]: run start_server.sh here

File: hugegraph-test/pom.xml [hunk 9]
```diff
@@ -12,11 +12,45 @@
     <artifactId>hugegraph-test</artifactId>
 
     <dependencies>
+        <!--jacoco report-aggregate just statistics it's directly dependent module,
```
Comment (zhoney) [inline hugegraph-test/pom.xml]: its

**Pull request merged**